### PR TITLE
deployments: use centos:centos7 instead of deployment-example image

### LINKED ISSUE #11256: should include various info in status [Conformance] [It]
```
• Failure [307.080 seconds]
deploymentconfigs
/data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:767
  with enhanced status
  /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:347
    should include various info in status [Conformance] [It]
    /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:346

    Expected error:
        <*errors.errorString | 0xc8200f5600>: {
            s: "timed out waiting for the condition",
        }
        timed out waiting for the condition
    not to have occurred

    /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:328
------------------------------
```

Seems that the deployment pods couldn't come up, resulting in the deployer timing out:
```
Oct  7 00:59:14.411: INFO: At 2016-10-07 00:54:19 -0400 EDT - event for deployment-simple-1-89t2i: {kubelet 172.18.1.159} FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "POD" with RunContainerError: "runContainer: An error occurred trying to connect: Post http://%2Fvar%2Frun%2Fdocker.sock/containers/create?name=k8s_POD.ca35dcd0_deployment-simple-1-89t2i_extended-test-cli-deployment-5noet-f4xon_19ace84f-8c4a-11e6-946d-0ed616a35903_0c542bd2: EOF"

Oct  7 00:59:14.411: INFO: At 2016-10-07 00:54:19 -0400 EDT - event for deployment-simple-1-jn16b: {kubelet 172.18.1.159} FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "POD" with RunContainerError: "InspectContainer: Cannot connect to the Docker daemon. Is the docker daemon running on this host?"
```
```
Oct  7 00:59:14.097: INFO: Running 'oc logs --namespace=extended-test-cli-deployment-5noet-f4xon --config=/tmp/extended-test-cli-deployment-5noet-f4xon-user.kubeconfig pod/deployment-simple-1-deploy'
Oct  7 00:59:14.397: INFO: --- pod deployment-simple-1-deploy logs
--> Scaling deployment-simple-1 to 2
--> Waiting up to 10m0s for pods in deployment deployment-simple-1 to become ready---
```

### LINKED ISSUE #11257: should rollback to an older deployment [Conformance] [It]
```
• Failure [336.958 seconds]
deploymentconfigs
/data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:767
  rolled back
  /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:617
    should rollback to an older deployment [Conformance] [It]
    /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:616

    Expected error:
        <*errors.errorString | 0xc8200f55c0>: {
            s: "timed out waiting for the condition",
        }
        timed out waiting for the condition
    not to have occurred

    /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:597
------------------------------
```
This seems to be an issue with our deployment example failing readiness probes randomly
```
Oct  7 01:09:04.459: INFO: At 2016-10-07 01:04:12 -0400 EDT - event for deployment-simple-2-otucj: {kubelet 172.18.1.159} Unhealthy: Readiness probe failed: Get http://172.17.0.5:8080/: dial tcp 172.17.0.5:8080: getsockopt: connection refused
```

### PR DESCRIPTION
**Pull request by mfojtik**
Closes #11256
Closes #11257

### CODE REVIEW

**Comment by kargakis** [comment]
[test]

**Comment by kargakis** [comment]
Should fix both https://github.com/openshift/origin/issues/11257 and https://github.com/openshift/origin/issues/11256

**Comment by kargakis** [comment]
LGTM

File: test/extended/testdata/deployments/failing-pre-hook.yaml [hunk 1]
```diff
@@ -21,7 +21,10 @@ spec:
         name: hook
     spec:
       containers:
-      - image: "docker.io/openshift/deployment-example:v1"
+      - image: "docker.io/centos:centos7"
+        command:
+          - /bin/sleep
```
Comment (kargakis) [inline test/extended/testdata/deployments/failing-pre-hook.yaml]: indentation is wrong

**Comment by mfojtik** [comment]
[test]

**Comment by kargakis** [comment]
[merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/11074 [merge]

**Pull request merged**